src/extension.ts

```diff
@@ -3558,6 +3559,44 @@ export async function activate(context: vscode.ExtensionContext) {
       new HintHoverProvider(treeDataProvider)
     )
   );
+
+  checkAndNotifyMissingCompileCommands();
+  context.subscriptions.push(
+    vscode.workspace.onDidChangeWorkspaceFolders(() => {
+      checkAndNotifyMissingCompileCommands();
+    })
```

Comment:
Why not add this function call in the already existing `vscode.workspace.onDidChangeWorkspaceFolders ` ?

We probably need to wrap that one to `context.subscriptions.push(...)`